crates/e2e/tests/e2e/services.rs

```diff
@@ -96,13 +97,46 @@ pub fn to_wei(base: u32) -> U256 {
 /// See:
 /// - Documentation: <https://docs.gnosis-safe.io/contracts/signatures#pre-validated-signatures>
 /// - Code: <https://github.com/safe-global/safe-contracts/blob/c36bcab46578a442862d043e12a83fec41143dec/contracts/GnosisSafe.sol#L287-L291>
-pub fn safe_prevalidated_signature(owner: H160) -> Bytes<Vec<u8>> {
+pub fn gnosis_safe_prevalidated_signature(owner: H160) -> Bytes<Vec<u8>> {
     let mut signature = vec![0; 65];
     signature[12..32].copy_from_slice(owner.as_bytes());
     signature[64] = 1;
     Bytes(signature)
 }
 
+/// Generate an owner signature for EIP-1271.
+///
+/// The Gnosis Safe uses off-chain ECDSA signatures from its owners as the
+/// signature bytes when validating EIP-1271 signatures. Specifically, it
+/// expects a signed EIP-712 `SafeMessage(bytes message)` (where `message` is
+/// the 32-byte hash of the data being verified).
+///
+/// See:
+/// - Code: <https://github.com/safe-global/safe-contracts/blob/c36bcab46578a442862d043e12a83fec41143dec/contracts/handler/CompatibilityFallbackHandler.sol#L66-L70>
```

Comment:
@MartinquaXD - I added some additional documentation here and tried to better explain how the signing works for EIP-1271.

I also moved the function into a shared module and detached it from the order hashing (to make it clear which part is Gnosis Safe signature, and which part is Cow Protocol specific order hashing).